### DISCUSSION TITLE
Question about Truestrike RGB keyboard light effect

### DISCUSSION THREAD

**Discussion by bquanlv** [BartoszCichecki/LenovoLegionToolkit#1087] [General]
Hello Hello, everyone. I posted this discussion mainly to get some help with the rgb keyboard problem.

The models targeted are those released after 2022 (including 2022). If you have a post-2022 model with an RGB keyboard, please see if you have the same problem, and I would like to talk about it.

We know that starting in 2022, models with rgb keyboards no longer rely on software developed by other companies for rgb light effect control. Lenovo calls it the Truestrike RGB Keyboard. We only need to install the software that comes with Lenovo to control it.

I am from China, the model I use is Y9000X 2022, and the corresponding overseas version is Legion Slim 7i 2022. In our country, only the Y9000K series (the highest end) will provide RGB keyboard, but my computer is more special, it is the Chinese version of Legion Slim 7i and has RGB keyboard.

I installed the Lenovo Legion Tookits.
The problem is light effects management, Lenovo management software provides six sets of options for RGB keyboard, you can choose to keep six light effects. Six groups of light effects can be switched by FN+ space.
My problem now is that my first group of light effects is the same as the sixth group, through the shortcut key combination switch, you can see the pop-up prompt to switch which group, the second to the fifth can be switched in order. Once you switch from five, whether it's through a shortcut key combination or a mouse click on the software, the result is the same.
It is worth mentioning that when I checked the pop-up prompt through the shortcut key combination switch, I found that after switching to the sixth group, the pop-up window would always display the first group. When I pressed the button more than ten times, I switched to the second group. At this time, the corresponding display on the LLT interface switched to the second group, and then switched smoothly to the third group. This is different from the mouse click switch, I don't know whether it is a software problem or a hardware problem, because I don't know the LLT code very well.

As for the hardware, I learned from the disassembly video that the driver board of Truestrike RGB is located below the battery, corresponding to the left side of the touchpad, and a square circuit board is placed in an inner slot to provide RGB support and control.
<img width="1271" alt="批注 2023-12-17 140501" src="https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/104878678/ccfa778e-e01b-48a6-91c9-fe3415ab2d77">

I searched the forum to check whether the domestic models with RGB keyboards of the same year (2022) have such problems, and only found problems such as hardware failure to recognize control.

When I got my computer back, everything was normal. After I reinstalled the professional version of Windows system, I used LLT or Lenovo official software, and such problems of light effect management occurred. I can still use five different light effects, though. However, the problem of abnormal switching of this light effect group makes me more entangled.

Thank you for reading here and hope to get your reply.

**Comment by BartoszCichecki** [comment]
I have heard once before of this issue on a Legion 7 (2022). It was resolved only after the keyboard was replaced under warranty - so it seems to be hardware related, or maybe keyboard RGB controller firmware issue. I don't know of anyway to fix it on your own.
